src/realm/table.cpp

```diff
@@ -291,6 +291,36 @@ void Table::insert_column_link(size_t col_ndx, DataType type, StringData name, T
     get_descriptor()->insert_column_link(col_ndx, type, name, target, link_type); // Throws
 }
 
+size_t Table::add_column_list(DataType type, StringData name, bool nullable)
+{
+    size_t ndx = get_column_count();
+    insert_column_list(ndx, type, name, nullable);
+    return ndx;
+}
+
+void Table::insert_column_list(size_t column_ndx, DataType type, StringData name, bool nullable)
+{
```

Comment:
The type of the column will be `type_Table`, and the subtables will have only one column with name "list"